src/safari-app-extension/AlpheiosReadingTools/AppDelegate.swift

```diff
@@ -210,56 +210,13 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         fullAS.setAttributes([.font: currentFont], range: fullASRange2)
     }
     
-    
-    func applicationWillTerminate(_ aNotification: Notification) {
-        // Insert code here to tear down your application
-    }
-    
     func windowWillReturnUndoManager(window: NSWindow) -> UndoManager? {
         // Returns the NSUndoManager for the application. In this case, the manager returned is that of the managed object context for the application.
         return persistentContainer.viewContext.undoManager
     }
     
     func applicationShouldTerminate(_ sender: NSApplication) -> NSApplication.TerminateReply {
-        // The code below is required to store all unsaved Core Data changes that are in memory but might not be saved to the hard drive yet. We do that before an application termination to make sure no in-memory changes will be lost.
-        let context = persistentContainer.viewContext
-        
-        if !context.commitEditing() {
-            NSLog("\(NSStringFromClass(type(of: self))) unable to commit editing to terminate")
-            return .terminateCancel
-        }
-        
-        if !context.hasChanges {
-            return .terminateNow
-        }
-        
-        do {
-            try context.save()
-        } catch {
-            let nserror = error as NSError
-            
-            // Customize this code block to include application-specific recovery steps.
-            let result = sender.presentError(nserror)
-            if (result) {
-                return .terminateCancel
-            }
-            
-            let question = NSLocalizedString("Could not save changes while quitting. Quit anyway?", comment: "Quit without saves error question message")
-            let info = NSLocalizedString("Quitting now will lose any changes you have made since the last successful save", comment: "Quit without saves error question info");
-            let quitButton = NSLocalizedString("Quit anyway", comment: "Quit anyway button title")
-            let cancelButton = NSLocalizedString("Cancel", comment: "Cancel button title")
-            let alert = NSAlert()
-            alert.messageText = question
-            alert.informativeText = info
-            alert.addButton(withTitle: quitButton)
-            alert.addButton(withTitle: cancelButton)
-            
-            let answer = alert.runModal()
-            if answer == .alertSecondButtonReturn {
-                return .terminateCancel
-            }
-        }
-        // If we got here, it is time to quit.
+        // Application is ready to terminate
```

Comment:
just cuious about the removal of code here. what changes make it no longer necessary?

Reply:
We've stopped writing changes to Core Data in AppDelegate so we don't need this code any more. Its purpose was to make sure that all unsaved changes are stored on app exit. Now we save user data in BackgroundScript only. This is the place where we have all the user-related logic and I think it would make sense for it to be responsible for storing data too.

Reply:
ok.